### ISSUE TITLE
Improve code coverage

### ISSUE THREAD

**Issue by jnyrup** [fluentassertions/fluentassertions#1823]
### Description

We now have automatic code coverage as part of the CI pipeline, posting the result to https://coveralls.io/github/fluentassertions/fluentassertions

At the time of writing our line coverage is 95.75% and branch coverage is 92.45%.
We do not seek 100% coverage at all costs as some parts might not/hardly be testable, but there are _clearly_ spots were it's easy to write the missing tests.

Ideally all code should be testable through the public API and shouldn't require any changes to the implementation.
If you think the core project needs changes, please discuss those changes here first.

**Comment by chvollm** [comment]
I looked through a lot of code. From what I see most uncovered is either boilerplate (GetEnumerator, etc.) or "ThisShouldn'tHappen".

**Comment by dennisdoomen** [comment]
You should be able to see what's missing from https://coveralls.io/github/fluentassertions/fluentassertions?branch=develop. You'll need to log on yourself. You can also run the build script locally or use an IDE like Rider to get coverage results.

**Comment by chvollm** [comment]
It might also be possible to improve coverage by removing unnecessary code.

**Comment by chvollm** [comment]
> You should be able to see what's missing from https://coveralls.io/github/fluentassertions/fluentassertions?branch=develop. You'll need to log on yourself. You can also run the build script locally or use an IDE like Rider to get coverage results.

I do use my Visual Studio for this. It works great, I just wanted to stress my findings. But I will dig deeper into the areas where it's possible.

**Comment by ITaluone** [comment]
How can this happen: 
https://github.com/fluentassertions/fluentassertions/blob/d53e8d996fcfb9141a89ebf8230115aebe5263fd/Src/FluentAssertions/Equivalency/Steps/DataSetEquivalencyStep.cs#L31-L34

If not should I remove this (also related to `DataColumEquivalencyStep` and `DataRowEquivalencyStep`)

**Comment by jnyrup** [comment]
Here's an example of how to hit that line

```cs
// Arrange
var subject = new
{
    DataSet = "foobar"
};

var expected = new
{
    DataSet = new DataSet()
};

// Act
Action act = () => subject.Should().BeEquivalentTo(expected);

act.Should().Throw<XunitException>()
    .WithMessage("Expected*System.Data.DataSet*found System.String*");
```

**Comment by ITaluone** [comment]
I have bend over to reach this line of code:
https://coveralls.io/builds/48537553/source?filename=Src%2FFluentAssertions%2FEventRaisingExtensions.cs#L138
but I can not set up a test to do this.

Is it dead code?

**Comment by dennisdoomen** [comment]
No, but the real issue is the duplication between the two `WithArgs` methods. Test `When_the_event_args_are_of_a_different_type_it_should_throw` covers the same condition for the overload taking taking a single predicate, but we don't have an equivalent test for the overload that takes a collection of predicates.
